metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -553,13 +559,33 @@ def _apply_workunit_processors(
     def get_workunits(self) -> Iterable[MetadataWorkUnit]:
         workunit_processors = self.get_workunit_processors()
         workunit_processors.append(AutoSystemMetadata(self.ctx).stamp)
-        return self._apply_workunit_processors(
+        # Process main workunits
+        main_stream = self._apply_workunit_processors(
             workunit_processors, auto_workunit(self.get_workunits_internal())
         )
+        yield from main_stream
+
+        # Process profiling workunits
+        yield from self._process_profiling_stage(workunit_processors)
+
+    def _process_profiling_stage(
+        self, processors: List[Optional[MetadataWorkUnitProcessor]]
+    ) -> Iterable[MetadataWorkUnit]:
+        """Process profiling stage if source supports it."""
+        if (
```

Comment:
This check looks odd. Humans better understanding non negated checks.
 Can't we say:
```
if isinstance(self, ProfilingCapable) and self.is_profiling_enabled_internal(): 
    ....
    while ...
       yield ...

return
```

Reply:
done. I was doing that to reduce the indentation. More indentation = problem in readability. But yeah that makes sense too